### PR TITLE
W-10848489: Added circular progress component

### PR DESCRIPTION
**Pull request by avgalon**
## Description
Added a circular progress with the ability to handle 2 circles and children within them. This component also replaces the radial progress and has backwards compatability.
image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Are you adding a new package?
- [ ] yes
- [x] no
```
CTC Output:
Case approved, merging is possible.
The case is available at: https://gus.lightning.force.com/lightning/r/Case/500EE00000V6ROkYAN/view
Updated at Sun Mar 20 11:47:19 UTC 2022
```

### CODE REVIEW

File: src/stories/CircularProgress.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  title: 'core/CircularProgress',
+  component: CircularProgress,
+  argTypes: {
+    color: { control: 'color' },
+  },
+};
+
+const Template = (args) => (
+  <>
+    <h1>Circular progress</h1>
```
Comment (distroysf) [inline src/stories/CircularProgress.stories.tsx:16]: each component should be in it's own Template

File: src/components/CircularProgress.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        </>
+      );
+    }
+
+    return circles;
+  };
+
+  return (
+    <Svg
+      width={2 * radius}
```
Comment (distroysf) [inline src/components/CircularProgress.js:53]: why is everything multiplied by 2?

File: src/components/CircularProgress.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  children,
+}) => {
+  const percentages = Array.isArray(percentage) ? percentage : [percentage];
+  const size = 2 * radius;
+  const getCircles = () => {
+    const circles = [];
+
+    for (let i = 0; i < percentages.length; i++) {
+      const innerRadius = radius - circlesGap * i - 0.5 * strokeWidth - 2;
+      const circumference = 2 * Math.PI * innerRadius;
```
Comment (distroysf) [inline src/components/CircularProgress.js:22]: why is everything multiplied by 2?

Comment (avgalon) [reply src/components/CircularProgress.js:22]: Not everything, just the circle perimeter and the circle wrapper. don't forget we get radius as a prop.

File: src/components/CircularProgress.js (hunk 2 above)
Comment (avgalon) [reply src/components/CircularProgress.js:53]: we need the complete with and we get radius

**Review by distroysf — APPROVED** [review]

**Comment by distroysf** [comment]
👍

**Pull request closed without merging**